lambdas/build-stac/utils/regex.py

```diff
@@ -28,12 +29,17 @@ def _calculate_month_range(datetime_obj: datetime) -> DATERANGE:
 
 
 def extract_dates(
-    filename: str, datetime_range: events.INTERVAL
+    item: events.RegexEvent
 ) -> Union[Tuple[datetime, datetime, None], Tuple[None, None, datetime]]:
     """
     Extracts start & end or single date string from filename.
     """
 
+    if (start := item.start_datetime) and (end := item.end_datetime):
+        return parse(start), parse(end), None
+    elif single := item.single_datetime:
+        return None, None, parse(single)
```

Comment:
Adding this code within the `utils/regex.py::extract_dates()` function expands the scope of the function a bit.  The docstring describes it as:

> Extracts start & end or single date string **from filename**.

Also, the file it's in is `regex.py`.  The code you're adding involves neither the filename or regex.

I recommend placing this logic outside of this function, back into `utils/stac.py`. something like:

```py
if item.start_datetime and item.end_datetime:
    start_datetime = parse(item.start_datetime)
    end_datetime = parse(item.end_datetime)
    single_datetime = None
else:
    start_datetime, end_datetime, single_datetime = regex.extract_dates(
        item.s3_filename, item.datetime_range
    )
```

I know that you've modified a number of tests to support your change, however I do think this would keep the codebase a bit more modular and also keep this PR smaller.

Reply:
that makes sense. thanks 🙇